src/wallet/spend.cpp

```diff
@@ -1141,6 +1129,16 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
            result.GetWaste(),
            result.GetSelectedValue());
 
+    // vouts to the payees
+    for (const auto& recipient : vecSend)
```

Comment:
would it make sense to preallocate the vouts before pushing?
```C++
    txNew.vout.reserve(txNew.vout.size() + vecSend.size());
```

Reply:
Trying to keep this a move-only refactor, so would prefer to leave it for now. I end up touching this code again in #28201; will make a note of this for that PR.

Reply:
Split it out to a new PR so that we don't forget it: https://github.com/bitcoin/bitcoin/pull/30093/files

Reply:
Rebased https://github.com/bitcoin/bitcoin/pull/30093 after this merge - it became trivial after your changes